src/lib/core/datetime/index.ts

```diff
@@ -6,26 +6,14 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {PlatformModule} from '@angular/cdk/platform';
 import {NgModule} from '@angular/core';
-import {DateAdapter} from './date-adapter';
 import {MAT_DATE_FORMATS} from './date-formats';
-import {NativeDateAdapter} from './native-date-adapter';
 import {MAT_NATIVE_DATE_FORMATS} from './native-date-formats';
+import {NativeDateModule} from '@angular/cdk/datetime';
```

Comment:
Since you're already making significant changes here, we should update this to look like our other `index.ts` files. Basically this should only re-export `public-api.ts`, re-exports from elsewhere should be in `public-api.ts`, and the `MatNativeDateModule` definition should be in a file called `datetime-module.ts` (just like how you structured it for the cdk)